src/main/java/org/apache/commons/lang3/Conversion.java

```diff
@@ -0,0 +1,1563 @@
+     */
+    public static int hexDigitMsb0ToInt(final char hexDigit) {
+        switch (hexDigit) {
+        case '0':
+            return 0x0;
+        case '1':
+            return 0x8;
+        case '2':
+            return 0x4;
+        case '3':
+            return 0xC;
+        case '4':
+            return 0x2;
+        case '5':
+            return 0xA;
+        case '6':
+            return 0x6;
+        case '7':
+            return 0xE;
+        case '8':
+            return 0x1;
+        case '9':
+            return 0x9;
+        case 'a':// fall through
+        case 'A':
+            return 0x5;
+        case 'b':// fall through
+        case 'B':
+            return 0xD;
+        case 'c':// fall through
+        case 'C':
+            return 0x3;
+        case 'd':// fall through
+        case 'D':
+            return 0xB;
+        case 'e':// fall through
+        case 'E':
+            return 0x7;
+        case 'f':// fall through
+        case 'F':
+            return 0xF;
+        default:
+            throw new IllegalArgumentException("Cannot interpret '" + hexDigit + "' as a hexadecimal digit");
+        }
+    }
+
+    /**
+     * <p>
+     * Converts a hexadecimal digit into binary (represented as boolean array) using the default
+     * (Lsb0) bit ordering.
+     * </p>
+     * <p>
+     * '1' is converted as follow: (1, 0, 0, 0)
+     * </p>
+     *
+     * @param hexDigit the hexadecimal digit to convert
+     * @return a boolean array with the binary representation of {@code hexDigit}
+     * @throws IllegalArgumentException if {@code hexDigit} is not a hexadecimal digit
+     */
+    public static boolean[] hexDigitToBinary(final char hexDigit) {
+        switch (hexDigit) {
+        case '0':
+            return FFFF.clone();
+        case '1':
+            return TFFF.clone();
+        case '2':
+            return FTFF.clone();
+        case '3':
+            return TTFF.clone();
+        case '4':
+            return FFTF.clone();
+        case '5':
+            return TFTF.clone();
+        case '6':
+            return FTTF.clone();
+        case '7':
+            return TTTF.clone();
+        case '8':
+            return FFFT.clone();
+        case '9':
+            return TFFT.clone();
+        case 'a':// fall through
+        case 'A':
+            return FTFT.clone();
+        case 'b':// fall through
+        case 'B':
+            return TTFT.clone();
+        case 'c':// fall through
+        case 'C':
+            return FFTT.clone();
+        case 'd':// fall through
+        case 'D':
+            return TFTT.clone();
+        case 'e':// fall through
+        case 'E':
+            return FTTT.clone();
+        case 'f':// fall through
+        case 'F':
+            return TTTT.clone();
+        default:
+            throw new IllegalArgumentException("Cannot interpret '" + hexDigit + "' as a hexadecimal digit");
+        }
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 174:215, 232:273.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.